tools/querytee/proxy_endpoint_test.go

```diff
@@ -350,6 +350,97 @@ func Test_ProxyEndpoint_Comparison(t *testing.T) {
 	}
 }
 
+func Test_ProxyEndpoint_LogSlowQueries(t *testing.T) {
+	scenarios := map[string]struct {
+		slowResponseThreshold    time.Duration
+		preferredResponseLatency time.Duration
+		secondaryResponseLatency time.Duration
+	}{
+		"responses are below threshold": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 0 * time.Millisecond,
+			secondaryResponseLatency: 0 * time.Millisecond,
+		},
+		"one response above threshold": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 0 * time.Millisecond,
+			secondaryResponseLatency: 101 * time.Millisecond,
+		},
+		"responses are both above threshold, but lower than threshold between themselves": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 101 * time.Millisecond,
+			secondaryResponseLatency: 150 * time.Millisecond,
+		},
+		"responses are both above threshold, and above threshold between themselves": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 101 * time.Millisecond,
+			secondaryResponseLatency: 202 * time.Millisecond,
+		},
```

Comment:
Would be good to add some cases for these scenarios but where the latencies are swapped (ie. secondary is faster than primary)